NebulaPatcher/Patches/Dynamic/UIStatisticsWindow_Patch.cs

```diff
@@ -30,5 +30,13 @@ public static void _OnClose_Postfix()
                 Multiplayer.Session.Network.SendPacket(new StatisticsRequestEvent(StatisticEvent.WindowClosed));
             }
         }
+
+        [HarmonyPrefix]
+        [HarmonyPatch(nameof(UIStatisticsWindow.AddStatGroup))]
+        public static bool AddStatGroup_Prefix(int __0, ProductionStatistics ___productionStat)
+        {
+            //Skip when StatisticsDataPacket hasn't arrived yet
+            return (__0 >= 0 && ___productionStat.factoryStatPool[__0] != null);
```

Comment:
Should `__0 < ___productionStat.factoryStatPool.Length` be checked as well?

Reply:
factoryStatPool is assigned in ``ProductionStatistics.Init(GameData)`` :  
``this.factoryStatPool = new FactoryProductionStat[this.gameData.factories.Length];``
and gameData.factories.Length should be equal or greater than the number of planets, so client and host should have the same factoryStatPool.Length
``__0``, PlanetData.factoryIndex that assigned in ``GameData.GetOrCreateFactory(PlanetData)``, is the index of the PlanetFactory of that planet in ``this.factories`` above 

Therefore ``planet.factoryIndex < ___productionStat.factoryStatPool.Length`` holds  as long as these mechanisms are not changed